packages/flutter/lib/src/widgets/animated_list.dart

```diff
@@ -316,7 +316,6 @@ class AnimatedListState extends State<AnimatedList> with TickerProviderStateMixi
   /// before [index] towards the beginning of the list.
   void removeItem(int index, AnimatedListRemovedItemBuilder builder, { Duration duration = _kDuration }) {
     assert(index != null && index >= 0);
-    assert(builder != null);
```

Comment:
this assertion should be modified to express that builder is only allowed to be null when duration is zero.